### PR TITLE
bsp: add ATSAMD21-XPRO eval board

### PR DESCRIPTION
**Pull request by majbthrd**
Add BSP for Atmel/Microchip [ATSAMD21-XPRO](https://www.microchip.com/DevelopmentTools/ProductDetails/ATSAMD21-XPRO)

### CODE REVIEW

**Comment by majbthrd** [comment]
In the ensuing months, I had forgotten that it would inevitably be doomed due to #255

**Comment by hathach** [comment]
@majbthrd can you re-open the PR, look like `adafruit/asf4` mass commenout the LITTLE_ENDIAN for all variants already https://github.com/adafruit/asf4/blob/circuitpython/samd21/include/samd21j18a.h

I will push the submodule update, please make a merge after that.

File: hw/bsp/board_mcu.h [hunk 1]
```diff
@@ -54,6 +54,7 @@
 
 #elif CFG_TUSB_MCU == OPT_MCU_SAMD11 || CFG_TUSB_MCU == OPT_MCU_SAMD21 || \
       CFG_TUSB_MCU == OPT_MCU_SAMD51 || CFG_TUSB_MCU == OPT_MCU_SAME5X
+  #undef LITTLE_ENDIAN // hack to suppress "LITTLE_ENDIAN" redefined
```
Comment (hathach) [inline hw/bsp/board_mcu.h:57]: there is no need to this, you only need to merge/rebase with the master.

File: hw/bsp/atsamd21_xpro/atsamd21_xpro.c [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ * This file is part of the TinyUSB stack.
+ */
+
+#include "bsp/board.h"
+
+#undef LITTLE_ENDIAN /* workaround for buggy SAMD submodule: compile error for ATSAMD21J18 (J, not G) */
```
Comment (hathach) [inline hw/bsp/atsamd21_xpro/atsamd21_xpro.c:29]: we could safely remove this one as well :)

**Review by hathach — APPROVED** [review]
it is perfect now, thanks for the PR.

**Pull request merged**